rundeckapp/src/main/groovy/org/rundeck/security/RundeckJaasAuthenticationSuccessEventListener.groovy

```diff
@@ -56,16 +57,12 @@ class RundeckJaasAuthenticationSuccessEventListener implements ApplicationListen
                 LOG.debug("Last Name: " + lastNamePrincipal?.name)
                 LOG.debug("Email: " + emailPrincipal?.name)
 
-                eventBus.
-                    notify(
-                        UserService.G_EVENT_LOGIN_PROFILE_CHANGE,
-                        new UserService.UserProfileData(
-                            username: username,
-                            lastName: lastNamePrincipal?.name,
-                            firstName: firstNamePrincipal?.name,
-                            email: emailPrincipal?.name
-                        )
-                    )
+                userService.updateUserProfile(
+                        username,
+                        lastNamePrincipal?.name,
+                        firstNamePrincipal?.name,
+                        emailPrincipal?.name
+                )
```

Comment:
why change this from the event style? the event is also sent by RundeckPreauthSuccessEventHandler, is there a reason to do it differently here?

Reply:
If the updateUserProfile is made here with the notify as it was before on the first login with LDAP a duplicate record will be generated on the RDUSER table since both transactions are happening at the same time, I  also had to add a Synchronized to the updateUserProfile method to avoid this behavior but wouldn't take effect if the notify is used here.

Reply:
then the same problem could happen with the Preauth handler?  I think that change only avoids the problem, perhaps we should add the unique constraint to the username

Reply:
Yes, I was just looking for a way to prevent this behavior since adding the constraint will cause an error in clients that already have duplicated login names on RDUSER table. I tested the proposed changes and worked to avoid the creation of duplicated users without causing any error in DB with duplicated records. If we want to add this constraint we need to be sure that the user DB doesn't have already a duplicated login name.